src/plugins/data/common/field_formats/types.ts

```diff
@@ -42,3 +42,23 @@ export interface FieldFormatConvert {
   text: TextContextTypeConvert;
   html: HtmlContextTypeConvert;
 }
+
+/** @public **/
+export enum FIELD_FORMAT_IDS {
```

Comment:
@Avinar-24 I'm probably missing something here, but how do we handle a situation where someone registers a new field format which has an ID that isn't in this enum?

For example, if I'm registering a new format, `FieldFormat` will let me specify any string as my ID, but when I go to `register` it fails due to the ID not being in the `FIELD_FORMAT_IDS` enum:
```ts
// myFormat.ts
export class MyFormat extends FieldFormat {
  static id = 'foo'; // TypeScript is okay with this...
  static title = 'Foo';
  ...etc
}

// myPlugin.ts
class MyPlugin {
  setup (core, { data }) {
    data.fieldFormats.register([ MyFormat ]); // ...But TypeScript doesn't like this
  }
}
```

Reply:
Fair enough. It should be possibly to create custom formatters. The type has been changed